### ISSUE TITLE
Support for GNU Makefiles: How can I define motions ]], [[, ][, [] to jump between rules?

### ISSUE THREAD

**Issue by kiryph** [nvim-treesitter/nvim-treesitter-textobjects#287]
An example Makefile
```make
edit : main.o kbd.o command.o display.o \
       insert.o search.o files.o utils.o
        cc -o edit main.o kbd.o command.o display.o \
                   insert.o search.o files.o utils.o

main.o : main.c defs.h
        cc -c main.c
kbd.o : kbd.c defs.h command.h
        cc -c kbd.c
command.o : command.c defs.h command.h
        cc -c command.c
display.o : display.c defs.h buffer.h
        cc -c display.c
insert.o : insert.c defs.h buffer.h
        cc -c insert.c
search.o : search.c defs.h buffer.h
        cc -c search.c
files.o : files.c defs.h buffer.h command.h
        cc -c files.c
utils.o : utils.c defs.h
        cc -c utils.c
clean :
        rm edit main.o kbd.o command.o display.o \
           insert.o search.o files.o utils.o
```

I have written this `textobjects.scm`:
```
(rule
  (recipe (_) @class.inner)?) @class.outer
```
However, I only get an error, when I press `]]`:
```
|| E5108: Error executing lua: ...ter-textobjects/lua/nvim-treesitter/textobjects/move.lua:53: bad argument #1 to 'ipairs' (table expected, got nil)
|| stack traceback:
|| 	[C]: in function 'ipairs'
|| 	...ter-textobjects/lua/nvim-treesitter/textobjects/move.lua:53: in function 'move'
|| 	...ter-textobjects/lua/nvim-treesitter/textobjects/move.lua:73: in function <...ter-textobjects/lua/nvim-treesitter/textobjects/move.lua:72>
|| 	...r-textobjects/lua/nvim-treesitter/textobjects/attach.lua:28: in function <...r-textobjects/lua/nvim-treesitter/textobjects/attach.lua:27>
```

### Filetype specific captures for `]]`, ...

I have a more general question in this context:

When I define in my init.lua following:
```lua
    move = {
      enable = true,
      set_jumps = true, -- whether to set jumps in the jumplist
      goto_next_start = {
        [']m'] = '@function.outer',
        [']]'] = '@class.outer',
      },
      goto_next_end = {
        [']M'] = '@function.outer',
        [']['] = '@class.outer',
      },
      goto_previous_start = {
        ['[m'] = '@function.outer',
        ['[['] = '@class.outer',
      },
      goto_previous_end = {
        ['[M'] = '@function.outer',
        ['[]'] = '@class.outer',
      },
    },
```

How can I say that I want to use different captures for different filetypes for section movement with `[[`, `]]`, `[]`, `][`?
Maybe `@block.outer` instead of `@class.outer`.

**Comment by kiryph** [comment]
Pull request https://github.com/nvim-treesitter/nvim-treesitter-textobjects/pull/286 seems to resolve the error here.

**Comment by theHamsta** [comment]
Can you try https://github.com/nvim-treesitter/nvim-treesitter-textobjects/compare/master...theHamsta:nvim-treesitter-textobjects:not-nil?expand=1

and please excuse the recent instability of this plugin

**Comment by theHamsta** [comment]
Fixed by #286

**Comment by kiryph** [comment]
However, I get following error for `][` (move to end of current @class.outer):
```
|| E5108: Error executing lua: ...r/start/nvim-treesitter/lua/nvim-treesitter/ts_utils.lua:431: Column value outside range
|| stack traceback:
|| 	[C]: in function 'nvim_win_set_cursor'
|| 	...r/start/nvim-treesitter/lua/nvim-treesitter/ts_utils.lua:431: in function 'goto_node'
|| 	...ter-textobjects/lua/nvim-treesitter/textobjects/move.lua:68: in function 'move'
|| 	...ter-textobjects/lua/nvim-treesitter/textobjects/move.lua:76: in function <...ter-textobjects/lua/nvim-treesitter/textobjects/move.lua:75>
|| 	...r-textobjects/lua/nvim-treesitter/textobjects/attach.lua:29: in function <...r-textobjects/lua/nvim-treesitter/textobjects/attach.lua:28>
```
